### PR TITLE
feat(elementhandle): add elementHandle.$$eval method

### PR DESCRIPTION
**Pull request by vasyl-pahut**
Fixes https://github.com/GoogleChrome/puppeteer/issues/2401

### CODE REVIEW

**Comment by vasyl-pahut** [comment]
I signed it!

File: docs/api.md [hunk 1]
```diff
@@ -2386,6 +2387,35 @@ The method runs `element.querySelector` within the page. If no element matches t
...
+<div class="feed">
+  <div class="tweet">
+    <div class="likes">100</div>
+    <div class="retweets">10</div>
+  </div>
+  <div class="tweet">
+    <div class="likes">10</div>
+    <div class="retweets">1</div>
+  </div>
+</div>
```
Comment (yanivefraim) [inline docs/api.md]: I would change the example to be simpler, with only one parent div + two children

File: lib/ElementHandle.js [hunk 2]
```diff
@@ -295,6 +295,25 @@ class ElementHandle extends JSHandle {
     return result;
   }
 
+  /**
+   * @param {string} selector
+   * @param {Function|String} pageFunction
+   * @param {!Array<*>} args
+   * @return {!Promise<(!Object|undefined)>}
+   */
+  async $$eval(selector, pageFunction, ...args) {
```
Comment (aslushnikov) [inline lib/ElementHandle.js:304]: Let's re-use this implementation in `frame.$$eval` method similarly to how it's done for `farme.$eval`:

https://github.com/GoogleChrome/puppeteer/blob/17bc6515deec8229afb73edc3869f497e6c06e10/lib/FrameManager.js#L368-L372

**Review by aslushnikov — APPROVED** [review]
Thanks!

**Pull request merged**